loki/transformations/pool_allocator.py

```diff
@@ -851,8 +851,6 @@ def create_pool_allocator(self, routine, stack_size):
                 _real_size_bytes = Cast(name='REAL', expression=Literal(1), kind=_kind)
                 _real_size_bytes = InlineCall(Variable(name='C_SIZEOF'),
                                               parameters=as_tuple(_real_size_bytes))
-                _real_size_bytes = InlineCall(function=Variable(name='MAX'),
```

Comment:
I haven't thought it through completely yet, but do we really never need to pad here?

Reply:
We pad the individual allocations themselves to ensure correct alignment on device. But here we are just converting the stack size back to bytes from integers. So we should never pad here and we should also never pad the denominator when we convert the stack size from bytes to integers (just before the actual stack allocation).